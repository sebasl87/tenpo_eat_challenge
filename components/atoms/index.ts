export { default as RestaurantsLayout } from './RestaurantsLayout';
