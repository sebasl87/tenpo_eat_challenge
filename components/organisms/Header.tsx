import React from 'react'
import { Image, Text, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.header}>
                <Image
                    source={require("../../assets/imgs/user.png")}
                    style={styles.userImg}
                />
                <Image
                    source={require("../../assets/icons/search.png")}
                    style={styles.searchImg}
                />
            </View>
            <View style={styles.header}>
                <View style={{ flexDirection: "column" }}>
                    <Image
                        source={require("../../assets/imgs/logo.png")}
                        style={styles.logoImg}
                    />
                    <Text>DELIVER APP</Text>
                </View>
                <View style={{ flexDirection: "row", marginBottom: -16 }}>
                    <Image
                        source={require("../../assets/imgs/hand.png")}
                        style={styles.handImg}
                    />
                    <View style={{ top: "100%", height: 6 }}>

                        <Image
                            source={require("../../assets/icons/ellipse.png")}
                            style={styles.ellipseImg}
                        />
                    </View>
                </View>
            </View>
        </>
    );
}

export default Header