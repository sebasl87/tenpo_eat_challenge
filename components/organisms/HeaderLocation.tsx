import React from 'react'
import { Image, Text, TextInput, View } from 'react-native';
import { styles } from '../../styles/styles';

function Header() {
    return (
        <>
            <View style={styles.headerLocation}>
                <View style={{ marginBottom: 32, display: "flex", flexDirection: "row" }}>
                    <Image
                        source={require("../../assets/icons/location.png")}
                        style={{ height: 24, width: 24, marginRight: 6, paddingTop: 30 }}
                    />
                    <Text style={styles.fontTitle}>Agregar dirección de la entrega</Text>
                </View>
            </View>
            <TextInput placeholder='Escribe tu dirección' style={styles.inputLocation} />
        </>
    );
}

export default Header