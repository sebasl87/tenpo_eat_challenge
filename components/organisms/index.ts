export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
