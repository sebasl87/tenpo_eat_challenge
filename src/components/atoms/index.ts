import AddressLayout from './AddressLayout';
export { default as RestaurantsLayout } from './RestaurantsLayout';
export { default as CategoriesLayout } from './CategoriesLayout';
export { default as HeaderFavorites } from './HeaderFavorites';
export { default as AddressLayout } from './AddressLayout';
