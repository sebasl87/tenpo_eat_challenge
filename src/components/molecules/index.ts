export { default as DetailCard } from './DetailCard';
export { default as DiscountCircle } from './DiscountCircle';
export { default as LogoRounded } from './LogoRounded';
export { default as DetailFavorites } from './DetailFavorites';
