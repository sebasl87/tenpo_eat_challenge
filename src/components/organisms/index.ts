export { default as Header } from "./Header";
export { default as HeaderLocation } from "./HeaderLocation";
export { default as ContainerMapView } from "./ContainerMapView";
export { default as RestaurantRow } from "./RestaurantRow";
export { default as ContainerAddressData } from "./ContainerAddressData";
export { default as CategoriesRow } from "./CategoriesRow";
export { default as FavoritesRow } from "./FavoritesRow";
export { default as HomeLocation } from './HomeLocation';
