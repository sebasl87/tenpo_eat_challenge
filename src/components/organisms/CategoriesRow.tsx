import React from 'react'
import { View, ScrollView, Text } from 'react-native';
import styled from 'styled-components/native';
import { CategoriesLayout } from '../atoms'

export interface ICategories { }

const Title = styled.Text`
  color: #333333;
  font-size: 18px;
  margin-bottom: 6px;
  margin-top: 32px;
`;

const linksData = [
    {
        uri: require("../../../assets/imgs/hamburguer.png"),
        category: "HAMBURGUESAS"
    },
    {
        uri: require("../../../assets/imgs/italian.png"),
        category: "ITALIANA"
    },
    {
        uri: require("../../../assets/imgs/pizza.png"),
        category: "PIZZAS"
    },

]

const CategoriesRow: React.FC<ICategories> = () => (
    <>

        <Title style={{ fontFamily: "Gotham-Bold" }}>CATEGORIAS</Title>
        <ScrollView horizontal={true}>
            <View style={{ flexDirection: "row" }}>
                {linksData.map((item, i) => (
                    <View key={i} style={{ marginRight: 6 }}>

                        <CategoriesLayout image={item.uri}>{item.category}</CategoriesLayout>
                    </View>
                ))}
            </View >
        </ScrollView>

    </>
);


export default CategoriesRow;