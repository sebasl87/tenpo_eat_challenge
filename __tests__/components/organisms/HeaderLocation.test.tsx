import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { HeaderLocation } from "../../../src/components/organisms";


jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn()
        .mockReturnValueOnce({
            google: { geocode: [], loading: false }, address: { address: { formated_address: "Some mock street, Santiago, Chile" } }, restaurants: {

                loading: false
            }
        }
        )
}));

jest.mock("@react-navigation/native", () => ({
    ...jest.requireActual("@react-navigation/native"),
    useNavigation: () => ({
        goBack: jest.fn(),
    }),

    useRoute: () => ({
        params: {
            resto: {
                id: 1,
                uri: "https://i.imgur.com/1PLcyBL.png",
                uri_large: "https://i.imgur.com/25zvKsX.png",
                discount: "50",
                restaurantName: "Mcdonalds",
                rate: "3.5",
                lapseTime: "10-50",
                bgColor: ""
            }
        },
    }),
}));



describe("HeaderLocation component", () => {
    it("should render successfully with mock", () => {
        const headerLocation = renderer
            .create(<HeaderLocation title="casa" />)
            .toJSON();

        expect(headerLocation).toMatchSnapshot();
    });

});