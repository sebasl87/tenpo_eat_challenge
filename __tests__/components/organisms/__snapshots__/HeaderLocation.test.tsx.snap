// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLocation component should render successfully with mock && length 1`] = `
<View
  style={
    {
      "height": 1200,
      "position": "absolute",
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-end",
        "backgroundColor": "#D4F9F5",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingTop": 80,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 32,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/location.png",
          }
        }
        style={
          {
            "height": 24,
            "marginRight": 6,
            "paddingTop": 30,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#2c8f7e",
            "fontFamily": "Roboto-light",
            "fontSize": 18,
            "paddingTop": 3,
          }
        }
      >
        casa
      </Text>
    </View>
  </View>
  <TextInput
    clearButtonMode="always"
    placeholder="Escribe tu dirección"
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 50,
        "color": "#2c8f7e",
        "fontFamily": "Roboto-light",
        "fontSize": 16,
        "minHeight": 56,
        "paddingLeft": 16,
        "paddingTop": 3,
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
        "top": -16,
      }
    }
  />
</View>
`;

exports[`HeaderLocation component should render successfully with mock 1`] = `
<View
  style={
    {
      "height": 1200,
      "position": "absolute",
      "width": "100%",
      "zIndex": 100,
    }
  }
>
  <View
    style={
      {
        "alignItems": "flex-end",
        "backgroundColor": "#D4F9F5",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingTop": 80,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 32,
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/location.png",
          }
        }
        style={
          {
            "height": 24,
            "marginRight": 6,
            "paddingTop": 30,
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#2c8f7e",
            "fontFamily": "Roboto-light",
            "fontSize": 18,
            "paddingTop": 3,
          }
        }
      >
        casa
      </Text>
    </View>
  </View>
  <TextInput
    clearButtonMode="always"
    placeholder="Escribe tu dirección"
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 50,
        "color": "#2c8f7e",
        "fontFamily": "Roboto-light",
        "fontSize": 16,
        "minHeight": 56,
        "paddingLeft": 16,
        "paddingTop": 3,
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
        "top": -16,
      }
    }
  />
</View>
`;
